### ISSUE TITLE
Feature request: -Wl,--force-exe-suffix counterpart

### ISSUE THREAD

**Issue by hujiajie** [emscripten-core/emscripten#14386]
I failed to find an option that could automatically append `.js` to the output of `emcc -o foobar foobar.c`.

In the `ld` man page:

```
       --force-exe-suffix
           Make sure that an output file has a .exe suffix.

           If a successfully built fully linked output file does not have a
           ".exe" or ".dll" suffix, this option forces the linker to copy the
           output file to one of the same name with a ".exe" suffix. This
           option is useful when using unmodified Unix makefiles on a
           Microsoft Windows host, since some versions of Windows won't run an
           image unless it ends in a ".exe" suffix.
```

Much appreciated if similar functionality can be provided in Emscripten.

**Comment by sbc100** [comment]
Interesting..  I've not seen than option before.   I notice this option does not exist in `lld`.. which makes me thing perhaps its is not commonly used.
 
Can you explain how/why exactly it would be helpful?  My understanding is that node will quite happily run a file called `foobar` (e.g. `node foobar`).  

If we were to add this, perhaps we would add it to emcc rather than wasm-ld.   In that case would it makes sense to call the options `--force-js-suffix` or should we stick with `--force-exe-suffix`?

**Comment by hujiajie** [comment]
I'm not sure if it's possible to create Win32 executables with clang + lld using gcc-compatible flags, maybe `-Wl,--force-exe-suffix` doesn't yet make much sense today. The `lld-link` command does produce Win32 executables, however it accepts MSVC link.exe flags, so most likely a `.exe` suffix is already there in the command line and `--force-exe-suffix` is still not that useful. Therefore, I feel the story of `lld` is slightly different from `ld`.

Un-suffixed JS/HTML output does work, but I guess `foobar.js`/`foobar.html` is more eye-candy to human, and I'd like to avoid patching 3rd-party Makefiles for this.

I don't suggest to reuse `--force-exe-suffix`. In the case of `--force-js-suffix`, then probably `--force-html-suffix` is also needed, or it's even better to have a single unified flag because emcc knows what is being emitted.

**Comment by sbc100** [comment]
I would imagine that the build system might get confused if make target did not exist.  i.e. if you do "make foo" and the the linker doesn't actually build anything with the name `foo` but instead builds `foo.js`.  

Possible issues include `make` itself not knowing that the project is up-to-date and always rebuilding/re-linking and also one part of the build system trying to run a program built earlier (it would get the name wrong).

Possible solution would be to create an executable shell script in this case which is called `foo` and can run `foo.js` under node?

**Comment by hujiajie** [comment]
Actually `--force-exe-suffix` creates both `foo` and `foo.exe`.

**Comment by sbc100** [comment]
Oh interesting.. are they the same file?

**Comment by hujiajie** [comment]
Yes. But `foo` has the executable bit set, the other not. Tested on Linux with MinGW-w64.
